kickstarter/android-oss
# MBL-1861: Fix survey notifications not working & MBL-1862: Add support for survey url to deeplinks

# 📲 What

Needed to add support for the new way we are receiving a survey url through push notifications, as well as add support for deeplinking to that url as well

# 🤔 Why

Users should be able to go to their survey from a push notification, and this means users were not receiving notifications for surveys

# 🛠 How

- Added new path format to the function that checks urls in `PushNotification.kt` as well as `DeeplinkViewModel.kt`
- This new path is `/projects/:creator_param/:project_param/backing/survey_responses`
- Since there are a few entry points for survey, in the `SurveyResponseViewModel.kt` we retrieve and format the url depending on where it comes from, and load it into the web view

# 👀 See


# 📋 QA

- Testing deeplinks to survey will only work on prod, so if you can, open survey link from email and make sure it's working
- Open a survey via the activity and make sure that is working
- To test push notifications on staging, you will have to add a survey url path to this function in `DebugPushNotificationsView.kt`:
```
fun simulateProjectSurveyButtonClick() {
        val gcm = GCM.builder()
            .title("Backer survey")
            .alert("Response needed! Get your reward for backing bugs in the office.")
            .build()
        val envelope: PushNotificationEnvelope = PushNotificationEnvelope.builder()
            .gcm(gcm)
            .survey(PushNotificationEnvelope.Survey.builder().id(18249859L).urls(PushNotificationEnvelope.Urls.builder().web(PushNotificationEnvelope.Web.builder().survey("/projects/:creator_param/:project_param/backing/survey_responses").build()).build())
                .projectId(PROJECT_ID)
                .build()
            )
            .build()
        pushNotifications?.add(envelope)
    }
```
- Open a survey from the activity feed
# Story 📖

[MBL-1861: Fix survey notifications not working](https://kickstarter.atlassian.net/browse/MBL-1861)
[MBL-1862: Add support for survey url to deeplinks](https://kickstarter.atlassian.net/browse/MBL-1862)
